uint16[] positions
int16[] correction
float64 step_delay
float64 serv_delay
